### ISSUE TITLE
unable to mount single usb device to the container when multiple usb devices are connected to the host machine in privileged mode 

### ISSUE THREAD

**Issue by vamsidhar5434** [moby/moby#35877]
I have multiple devices(android devices) connected to my docker host, I am trying to run the mobile tests in parallel on these devices. what i am trying do is access these devices on different containers and route particular tests to different containers which are connected to a unique single android device but I was unable to mount a single device on to a particular docker container in privilege mode (instead all devices are accessible inside a container)

Steps to Reproduce the issue:

1. connect multiple android devices and put the in file transfer mode 
2. create a container by mounting usb devices like 
    docker run -it --name MVE1 --privileged -v /dev/bus/usb/001/002:/dev/bus/usb/001/002 dockerimage (trying to connect 002 device alone to the new container )

Expected : 
The container created with the name MVE1 will have only one device connected irrespective of how many devices connected to the host 
run using shell :  adb devices  
only one device should display not all devices which are connected to host 

Actual:
All usb devices are accessed in the MVE1 contained which is not expected when only one device is mounting when creating a container
run using shell : adb devices 
all the devices which are connected to host are listed which is not expected 

Output of `docker version:

```
Client:
 Version:      17.09.1-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   19e2cf6
 Built:        Thu Dec  7 22:23:40 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.1-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   19e2cf6
 Built:        Thu Dec  7 22:25:03 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 6
 Running: 1
 Paused: 0
 Stopped: 5
Images: 91
Server Version: 17.09.1-ce
Storage Driver: devicemapper
 Pool Name: docker-253:0-786448-pool
 Pool Blocksize: 65.54kB
 Base Device Size: 10.74GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 11.32GB
 Data Space Total: 107.4GB
 Data Space Available: 13.34GB
 Metadata Space Used: 14.5MB
 Metadata Space Total: 2.147GB
 Metadata Space Available: 2.133GB
 Thin Pool Minimum Free Space: 10.74GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.140-RHEL7 (2017-05-03)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-693.5.2.el7.x86_64
Operating System: OpenShift Enterprise
OSType: linux
Architecture: x86_64
CPUs: 32
Total Memory: 62.89GiB
Name: mobilevalab.rpega.com
ID: EB35:UAKF:HCYP:AFCQ:X2KS:RJNY:E5UL:ISAT:6HSW:GLST:CE2C:DTAE
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: devicemapper: usage of loopback devices is strongly discouraged for production use.
         Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Comment by Febin-kg** [comment]
I'm stuck with this issue too.
Any workarounds ?

**Comment by vamsidhar5434** [comment]
Remove the remaining device block files from container except one device you need to connect, 
surprisingly this will not remove the block file from host system but only from the container. I am handling this logic in docker entry point so that we need not delete unwanted devices manually every time

**Comment by Febin-kg** [comment]
I manually removed it from the container and the file was removed from the host as well.
Could you please share your Dockerfile content to see how you have handled the entrypoint ?

**Comment by Febin-kg** [comment]
I've run it using the command `docker run --privileged -v /dev/bus/usb:/dev/bus/usb -d test`. Next, I exec into the running container and perform `rm /dev/bus/usb/002/005`. This deletes the file from both the container as well as the host. I would like to retain the file in the host.
Appreciate the help.

**Comment by vamsidhar5434** [comment]
/dev/bus/usb:/dev/bus/usb will mount every device connected to the any usb bus in the host. 
If you want to mount a device 004 connected to 002 bus try this "docker run --privileged -v /dev/bus/usb/002/004:/dev/bus/usb/002/004 -d image_name" even if this command mounts all devices try to remove other devices from contains and see it retains in the host

**Comment by Febin-kg** [comment]
That removes it from the host as well.
In case anyone else is stuck with this same issues,[ this SO post ](https://stackoverflow.com/questions/50851990/using-docker-run-privileged-to-host-a-single-usb-device/50861123) helped.

**Comment by vamsidhar5434** [comment]
I have tried with --device option but adb is not recognising the device, so I removed the usb file in my case it's not removing from host
